core/src/main/java/hudson/security/GlobalSecurityConfiguration.java

```diff
@@ -57,38 +66,84 @@
+        return getJenkins().getMarkupFormatter();
     }
 
     public int getSlaveAgentPort() {
-        return Jenkins.getInstance().getSlaveAgentPort();
+        return getJenkins().getSlaveAgentPort();
     }
 
     public boolean isDisableRememberMe() {
-        return Jenkins.getInstance().isDisableRememberMe();
+        return getJenkins().isDisableRememberMe();
+    }
+
+    public boolean isUseSecurity() {
+        return getJenkins().isUseSecurity();
+    }
+
+    public SecurityRealm getSecurityRealm() {
+        return getJenkins().getSecurityRealm();
+    }
+
+    public AuthorizationStrategy getAuthorizationStrategy() {
+        return getJenkins().getAuthorizationStrategy();
+    }
+
+    /**
+     * This is a method to test authentication configuration, it configures a
+     * transient Jenkins instance and uses that to invoke configuration methods
+     */
+    public synchronized void doTestAuthentication(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, FormException {
+        SecurityContext oldContext = ACL.impersonate(ACL.SYSTEM);
+        try {
+            Jenkins wiz = new TransientJenkins(Jenkins.getInstance());
+            configure(wiz, req, req.getSubmittedForm());
+            SecurityRealm securityRealm = wiz.getSecurityRealm();
+            if(securityRealm instanceof AbstractPasswordBasedSecurityRealm) {
+                String username = req.getParameter("username");
+                String password = req.getParameter("password");
+                ((AbstractPasswordBasedSecurityRealm) securityRealm).authenticate(username, password);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            SecurityContextHolder.setContext(oldContext);
+        }
     }
 
+    /**
+     * This is the primary configuration method, which applies directly to the global
+     * Jenkins instance
+     */
     public synchronized void doConfigure(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, FormException {
         // for compatibility reasons, the actual value is stored in Jenkins
-        BulkChange bc = new BulkChange(Jenkins.getInstance());
+        BulkChange bc = new BulkChange(getJenkins());
         try{
-            boolean result = configure(req, req.getSubmittedForm());
+            boolean result = configure(getJenkins(), req, req.getSubmittedForm());
             LOGGER.log(Level.FINE, "security saved: "+result);
-            Jenkins.getInstance().save();
+            getJenkins().save();
             FormApply.success(req.getContextPath()+"/manage").generateResponse(req, rsp, null);
         } finally {
             bc.commit();
         }
     }
 
-    public boolean configure(StaplerRequest req, JSONObject json) throws hudson.model.Descriptor.FormException {
+    protected boolean configure(Jenkins j, StaplerRequest req, JSONObject json) throws hudson.model.Descriptor.FormException {
```

Comment:
How about creating an implementation of `AccessControlled` instead of the `TransientJenkins`?